### PR TITLE
Add Dreamstate module

### PR DESCRIPTION
**Pull request by bilalm2**

### CODE REVIEW

File: src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { formatNumber } from 'common/format';
+
+const cdrPerTick = 3;
+
+class Dreamstate extends Analyzer {
+  tickCount = 0;
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_DRUID.GROVE_TENDING_TALENT);
```
Comment (trevorm4) [inline src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]:
```suggestion
    this.active = this.selectedCombatant.hasTalent(TALENTS_DRUID.GROVE_TENDING_TALENT);
    if (!this.active) { 
    		return;
    }
```

Comment (kfinch) [reply src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]: I must fight the good fight! Doing this is optional! (I don't do this, and the listeners won't be added even if you don't return)

File: src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import Events, { CastEvent } from 'parser/core/Events';
+import Statistic from 'parser/ui/Statistic';
+import { SpellIcon } from 'interface';
+import BoringValue from 'parser/ui/BoringValueText';
+import { formatNumber } from 'common/format';
+
+const cdrPerTick = 3;
```
Comment (kfinch) [inline src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]: All times and durations in both log and WoWA are in milliseconds, so this should be 3000.

The one exception is cooldowns as defined in the Abilities file, and I know this is kinda close to that but `SpellUsable` is going to want milliseconds so I'd prefer this was in millis too.

File: src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  statistic() {
+    return (
+      <Statistic
+        size="flexible"
+        position={STATISTIC_ORDER.OPTIONAL(6)} // number based on talent row
+        category={STATISTIC_CATEGORY.TALENTS}
+        tooltip={
+          <>
+            Total Cooldown Reduction from all Tranquility casts:{' '}
+            <strong>{this.totalCDR} seconds</strong>
```
Comment (kfinch) [inline src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]: Once totalCDR is converted to milliseconds, you can make this show correctly as `(this.totalCDR / 1000).toFixed(0)`

File: src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx (hunk 1 above)
Comment (trevorm4) [reply src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]: Technically it prevents some function calls for slight performance boost (lol) but not strictly necessary ig

File: src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        }
+      >
+        <BoringValue
+          label={
+            <>
+              <SpellIcon id={TALENTS_DRUID.DREAMSTATE_TALENT.id} /> Dreamstate CDR per minute
+            </>
+          }
+        >
+          <>{formatNumber(this.owner.getPerMinute(this.totalCDR))} seconds</>
```
Comment (kfinch) [inline src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]: `getPerMinute` already expects milliseconds so once you change `this.totalCDR` to millis, no change will be needed here

**Review by kfinch — COMMENTED** [review]
Looks good, just needs the SpellUsable thing we discussed in DMs and a changelog entry

File: src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx (hunk 1 above)
Comment (kfinch) [reply src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]: But it's not pretty reeeeeee

File: .env.local.example [hunk 5]
```diff
@@ -1,3 +0,0 @@
-# Copy this file and rename it to `.env.local` to configure your own WCL API key. You can get your own API key from the bottom of the Warcraft Logs settings: https://www.warcraftlogs.com/profile
-# You should also enter the application name "WoWAnalyzer (development)" on that page, just above the key.
-REACT_APP_WCL_API_KEY=INSERT_YOUR_OWN_API_KEY_HERE
```
Comment (kfinch) [inline .env.local.example]: Looks like you accidentally removed this when setting up your local. Please restore it.

File: src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx [hunk 6]
```diff
@@ -51,7 +59,7 @@ class Dreamstate extends Analyzer {
             </>
           }
         >
-          <>{formatNumber(this.owner.getPerMinute(this.totalCDR))} seconds</>
+          <>{formatNumber(this.owner.getPerMinute(this.totalCDR) / 1000)} seconds</>
```
Comment (kfinch) [inline src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]: You don't need to divide by 1000. `getPerMinute` already expects input in milliseconds.

Comment (kfinch) [reply src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]: Also would like to see a screenshot of how this looks

File: src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx (hunk 4 above)
Comment (kfinch) [reply src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]: for anyone else reading this - I am an idiot and `getPerMinute` actually takes 'unitless', not milliseconds

File: src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx (hunk 6 above)
Comment (kfinch) [reply src/analysis/retail/druid/restoration/modules/spells/Dreamstate.tsx]: for anyone else reading this - I am an idiot and `getPerMinute` actually takes 'unitless', not milliseconds. He does actually need to divide by 1000

**Review by kfinch — APPROVED** [review]

**Pull request merged**